src/bun.js/node-dns.exports.js

```diff
@@ -535,15 +548,21 @@ export const promises = {
     }
 
     resolve(hostname, rrtype) {
-      return dns.resolve(hostname, rrtype).then(promisifyResolve);
+      return dns.resolve(hostname, rrtype).then(promisifyResolve(rrtype));
     }
 
     resolve4(hostname, options) {
-      return dns.lookup(hostname, { family: 4 });
+      if (options?.ttl) {
+        return dns.lookup(hostname, { family: 4 });  
+      }
+      return dns.lookup(hostname, { family: 4 }).then((res)=> res?.map(a => a.address))
```

Comment:
can we generate each function at the top to avoid the allocation

also make sure you have Prettier configured to run on save (you might need to run `bun install` somewhere)

Reply:
done in https://github.com/oven-sh/bun/pull/2144/commits/9aeb64f0fe675a8382320842a5df8ba5f6e906a9